qt/scientific_interfaces/Indirect/FqFitModel.cpp

```diff
@@ -15,344 +15,21 @@
-    }
-  } else
-    logger.warning("Invalid width index specified.");
-}
-
-void FqFitModel::setActiveEISF(std::size_t eisfIndex, WorkspaceID workspaceID, bool single) {
-  const auto parametersIt = findFqFitParameters(workspaceID);
-  if (parametersIt != m_fqFitParameters.end() && parametersIt->second.eisfSpectra.size() > eisfIndex) {
-    const auto &eisfSpectra = parametersIt->second.eisfSpectra;
-    if (single == true) {
-      setSpectra(createSpectra(std::vector<std::size_t>({eisfSpectra[eisfIndex]})), workspaceID);
-    } else { // In multiple mode the spectra needs to be appending on the
-             // existing spectra list.
-      auto spectra_vec = std::vector<std::size_t>({eisfSpectra[eisfIndex]});
-      auto spectra = getSpectra(workspaceID);
-      for (auto i : spectra) {
-        if ((std::find(spectra_vec.begin(), spectra_vec.end(), i.value) == spectra_vec.end())) {
-          spectra_vec.push_back(i.value);
-        }
-      }
-      setSpectra(createSpectra(spectra_vec), workspaceID);
-    }
-  } else
-    logger.warning("Invalid EISF index specified.");
-}
-
-bool FqFitModel::isMultiFit() const {
-  if (getNumberOfWorkspaces() == WorkspaceID{0})
-    return false;
-  return !allWorkspacesEqual(getWorkspace(WorkspaceID{0}));
-}
-
-std::vector<std::string> FqFitModel::getWidths(WorkspaceID workspaceID) const {
-  const auto parameters = findFqFitParameters(workspaceID);
-  if (parameters != m_fqFitParameters.end())
-    return parameters->second.widths;
-  return std::vector<std::string>();
-}
-
-std::vector<std::string> FqFitModel::getEISF(WorkspaceID workspaceID) const {
-  const auto parameters = findFqFitParameters(workspaceID);
-  if (parameters != m_fqFitParameters.end())
-    return parameters->second.eisf;
-  return std::vector<std::string>();
-}
-
-boost::optional<std::size_t> FqFitModel::getWidthSpectrum(std::size_t widthIndex, WorkspaceID workspaceID) const {
-  const auto parameters = findFqFitParameters(workspaceID);
-  if (parameters != m_fqFitParameters.end() && parameters->second.widthSpectra.size() > widthIndex)
-    return parameters->second.widthSpectra[widthIndex];
-  return boost::none;
-}
-
-boost::optional<std::size_t> FqFitModel::getEISFSpectrum(std::size_t eisfIndex, WorkspaceID workspaceID) const {
-  const auto parameters = findFqFitParameters(workspaceID);
-  if (parameters != m_fqFitParameters.end() && parameters->second.eisfSpectra.size() > eisfIndex)
-    return parameters->second.eisfSpectra[eisfIndex];
-  return boost::none;
-}
+FqFitModel::FqFitModel() { m_fitType = FQFIT_STRING; }
 
 std::string FqFitModel::getResultXAxisUnit() const { return ""; }
 
 std::string FqFitModel::getResultLogName() const { return "SourceName"; }
 
-bool FqFitModel::allWorkspacesEqual(const Mantid::API::MatrixWorkspace_sptr &workspace) const {
-  for (auto i = WorkspaceID{1}; i < getNumberOfWorkspaces(); ++i) {
-    if (getWorkspace(i) != workspace)
-      return false;
-  }
-  return true;
-}
-
```

Comment:
Most of FqFitModel has been removed. Is there a reason to still have a separate FqFitModel?

Reply:
The three functions remaining are required to be defined seperatly. so we do still need FqFitModel

Reply:
Fair enough! Thought I would double check to be on the safe side.